tangelo/linq/circuit.py

```diff
@@ -415,17 +443,80 @@ def remove_small_rotations(circuit, param_threshold=0.05):
     rot_gates = {"RX", "RY", "RZ", "CRX", "CRY", "CRZ"}
     gates = [g for g in circuit._gates if not (g.name in rot_gates and abs(g.parameter) % (2*np.pi) < param_threshold)]
 
-    return Circuit(gates)
+    return Circuit(gates) if remove_qubits else Circuit(gates, n_qubits=circuit.width)
+
+
+def merge_rotations(circuit: Circuit):
+    """ Merge rotation gates successively applied to the same qubits (targets, controls), into a single
+    equivalent rotation gate. If one of the gates involved is labeled variational, the merged rotation is
+    as well.
+
+    Agglomerating non-numerical values for parameters is currently unsupported.
+
+    Args:
+        circuit (Circuit): Input quantum circuit
+
+    Returns:
+        Circuit: circuit with merged rotations
+    """
+
+    NoneGate = Gate('NONE', 0)
+
+    gate_qubits = {i: list() for i in range(circuit.width)}
+    new_gates = []
+
+    rot_gates = {"RX", "RY", "RZ", "CRX", "CRY", "CRZ"}
```

Comment:
We could add PHASE gate as well (also in remove_small_rotations - maybe a function name change would be necessary).

Reply:
To clarify: are you asking me to try to leverage the identities

$T^2 = S$ and $S^2 = Z$ ?

Reply:
Yep, it is the more "general" Phase gate, and we list it in the parametrized gates: https://github.com/goodchemistryco/Tangelo/blob/59f5cb9fb4ae59f0d2bd5991c038a9f00b05d997/tangelo/linq/gate.py#L34. I would guess we could also remove/merge those, as it is the same as performing a rotation in a single-qubit Bloch sphere.

Reply:
I have added PHASE and CPHASE and left XX out of scope for now, as Olivia's team knows IONQ very well and I'd rather have their take on it and leave opportunities for them to contribute :)